src/components/Profile/Profile.jsx

```diff
@@ -27,16 +54,179 @@ export default function Profile(props) {
     }
   }
 
+  // continue editing diets so secondary diet options don't show primary diet
+  useEffect(() => {
+    setDiets(primDiets.filter(val => {
+        return val.toLowerCase() != userPrimDiet
+    }))
+  }, [userPrimDiet])
+
+  function showPrimDietForm() {
+    if (showUserPrimDietForm) {
+      // get clicked data
+      const newUserPrimDiet = primDietChecked.map(val => 
+        val == "Gluten Free" ? "gluten-free" : val.toLowerCase()
+      )
+      setUserPrimDiet(newUserPrimDiet)
+      const docRef = doc(db, "users", currentUser.uid)
+        updateDoc(docRef, { primDiet: newUserPrimDiet })
+        .catch(error => {
+            setError(error.message)
+      })
+      setPrimDietChecked([])
+    }
+    setShowUserPrimDietForm(!showUserPrimDietForm)
+  }
+
+  function showDietsForm() {
```

Comment:
It seems a lot of your state is very similar in nature. This is why you have very similar handlers to deal with it. For example UserPrimDiet, UserDiets, UserAllergies all have a show function that basically does the same thing. In theory you can hold all of these in a map like { primDiet : { show: false, value: [], userDiets: { show: false, value: []}, etc... Then you can define a single common function such as show to deal with all of them, which takes the name of the key in the map. Even in your display code you can just iterate over the map to render all the fields. There are many benefits to this... code would be concise, so in theory easier to read, easier to fix bugs, since you only have to fix them in one place, easier to add new types of form fields.. just add to the map. This is a much more complex change, so I don't expect you to take that up at this time. Just wanted to point it out. If you'd like we can also have a quick VC to talk about it in person :)